core/src/main/scala/org/apache/spark/ExecutorAllocationManager.scala

```diff
@@ -455,7 +461,12 @@ private[spark] class ExecutorAllocationManager(
     val executorsRemoved = if (testing) {
       executorIdsToBeRemoved
     } else {
-      client.killExecutors(executorIdsToBeRemoved)
+      // We don't want to change our target number of executors, because we already did that
+      // when the task backlog decreased.  Normally there wouldn't be any tasks running on these
+      // executors, but maybe the scheduler *just* decided to run a task there -- in that case,
```

Comment:
Also don't follow this part. If `force = false`, doesn't that mean the executor won't be killed if tasks are running on it? So wouldn't `countFailures` be meaningless in that context?

Reply:
good point,  I didn't look closely enough at the semantics of `force`